ext/storage_sources/local_store/local_store.c

```diff
@@ -917,7 +919,8 @@ local_open(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
 
     if (0) {
 err:
-        local_file_close_internal(local, session, local_fh, true);
+        if (local_fh != NULL)
```

Comment:
Failed `calloc` can cause us to jump here. So we should check that the allocation actually worked to avoid a close here.